catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -631,34 +644,17 @@ public MetacardType getMetacardType(SolrDocument doc) throws MetacardCreationExc
     }
 
     byte[] bytes = (byte[]) doc.getFirstValue(SchemaFields.METACARD_TYPE_OBJECT_FIELD_NAME);
-
-    ByteArrayInputStream bais = null;
-    ObjectInputStream in = null;
     try {
-
-      bais = new ByteArrayInputStream(bytes);
-
-      in = new ObjectInputStream(bais);
-
-      cachedMetacardType = (MetacardType) in.readObject();
-
+      cachedMetacardType = OBJECT_MAPPER.readValue(bytes, MetacardType.class);
```

Comment:
❓ Won't that cause problems with databases created prior to this version where the info was serialized using Java and not Json? Should we detect if it is a Java serialized blob (i.e. starts with 0xCAFEBABE) or a Json string? 
Should we also consider adding a signature at the beginning of the byte buffer after Json serialization to detect the data was serialized using Json?

Reply:
#3553 related to DDF-3938 already requires all data to be reindexed.  It is common practice currently to dump and reindex data during upgrades.  This allows plugins and transformers to reprocess data as well.